lib/request_store_rails/railtie.rb

```diff
@@ -6,9 +6,15 @@ class Railtie < ::Rails::Railtie
 
     initializer 'request_store_rails.insert_middleware' do |app|
       app.config.middleware.insert_after ActionDispatch::RequestId, RequestStoreRails::Middleware
-
-      ActionDispatch::Reloader.to_cleanup do
-        RequestLocals.clear_all!
+     
+      if ActiveSupport.const_defined?(:Reloader) && ActiveSupport::Reloader.respond_to?(:to_complete)
+        ActiveSupport::Reloader.to_complete do
+          RequestLocals.clear_all!
+        end
+      elsif ActionDispatch.const_defined?(:Reloader) && ActionDispatch::Reloader.respond_to?(:to_cleanup)
+        ActionDispatch::Reloader.to_cleanup do
+          RequestLocals.clear_all!
+        end
```

Comment:
Sure. FYI this PR was a "cherry pick" of this commit : https://github.com/steveklabnik/request_store/commit/43298a51c004575758813eedb1b914ec5cd60c33